clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -267,6 +286,7 @@ protected void onJoinPrepare(int generation, String memberId) {
                     listener.getClass().getName(), groupId, e);
         }
 
+        assignmentSnapshot = null;
```

Comment:
It gets set in `performAssignment` which gets called before `onJoinComplete` for the leader.